drivers/input/input_stmpe811.c

```diff
@@ -0,0 +1,422 @@
+				     STMPE811_SYS_CTRL2_BIT_ADC_FCT, 0);
+	if (err < 0) {
+		return err;
+	}
+
+	/* Select sample time, bit number and ADC reference */
+	err = i2c_reg_write_byte_dt(&config->bus, STMPE811_ADC_CTRL1_REG, STMPE811_ADC_CTRL1_CONF);
+	if (err < 0) {
+		return err;
+	}
+
+	/* Select the ADC clock speed */
+	err = i2c_reg_write_byte_dt(&config->bus, STMPE811_ADC_CTRL2_REG, STMPE811_ADC_CLOCK_SPEED);
+	if (err < 0) {
+		return err;
+	}
+
+	/* Touch screen configuration */
+	err = i2c_reg_write_byte_dt(&config->bus, STMPE811_TSC_CFG_REG,
+				    stmpe811_tsc_config_bits(dev));
+	if (err < 0) {
+		return err;
+	}
+
+	/* Configure the touch FIFO threshold */
+	err = i2c_reg_write_byte_dt(&config->bus, STMPE811_FIFO_TH_REG,
+				    STMPE811_FIFO_TH_SINGLE_POINT);
+	if (err < 0) {
+		return err;
+	}
+
+	/* Clear the FIFO memory content */
+	err = i2c_reg_write_byte_dt(&config->bus, STMPE811_FIFO_STA_REG, STMPE811_FIFO_STA_CLEAR);
+	if (err < 0) {
+		return err;
+	}
+
+	/* Set the range and accuracy of the pressure measurement (Z) */
+	err = i2c_reg_write_byte_dt(&config->bus, STMPE811_TSC_FRACT_XYZ_REG,
+				    STMPE811_TSC_FRACT_XYZ_CONF);
+	if (err < 0) {
+		return err;
+	}
+
+	/* Set the driving capability (limit) of the device for TSC pins */
+	err = i2c_reg_write_byte_dt(&config->bus, STMPE811_TSC_I_DRIVE_REG,
+				    STMPE811_TSC_I_DRIVE_LIMIT);
+	if (err < 0) {
+		return err;
+	}
+
+	/* Touch screen control configuration */
+	err = i2c_reg_write_byte_dt(&config->bus, STMPE811_TSC_CTRL_REG,
+				    stmpe811_tsc_control_bits(dev));
+	if (err < 0) {
+		return err;
+	}
+
+	/**
+	 * Clear all the status pending bits if any.
+	 * Writing '1' to this register clears the corresponding bits.
+	 * This is an 8-bit register, so writing 0xFF clears all.
+	 */
+	err = i2c_reg_write_byte_dt(&config->bus, STMPE811_INT_STA_REG, BIT_MASK(8));
+	if (err < 0) {
+		return err;
+	}
+
+	/* Put the FIFO back into operation mode */
+	err = i2c_reg_write_byte_dt(&config->bus, STMPE811_FIFO_STA_REG,
+				    STMPE811_FIFO_STA_OPERATIONAL);
+	if (err < 0) {
+		return err;
+	}
+
+	/* Enable FIFO and touch interrupts */
+	err = i2c_reg_update_byte_dt(&config->bus, STMPE811_INT_EN_REG, BIT_MASK(8),
+				     STMPE811_INT_BIT_TOUCH | STMPE811_INT_BIT_FIFO_THRESHOLD);
+	if (err < 0) {
+		LOG_ERR("Could not enable interrupt types (%d)", err);
+		return err;
+	}
+
+	return 0;
+}
+
+static int stmpe811_ts_get_data(const struct device *dev)
+{
+	const struct stmpe811_config *config = dev->config;
+	struct stmpe811_data *data = dev->data;
+
+	uint8_t data_xy[3];
+	uint32_t uldata_xy;
+
+	int ret = i2c_burst_read_dt(&config->bus, STMPE811_TSC_DATA_NON_INC_REG, data_xy,
+				    sizeof(data_xy));
+	if (ret < 0) {
+		return ret;
+	}
+
+	/* Calculate positions values */
+	uldata_xy = (data_xy[0] << 16) | (data_xy[1] << 8) | data_xy[2];
+	data->touch_x = (uldata_xy >> 12U) & BIT_MASK(12);
+	data->touch_y = uldata_xy & BIT_MASK(12);
```

Comment:
Cool, though this is the raw adc reading right? Means it's not directly usable by, for example, LVGL. Should there be some calibration scaling to map this to pixels here?

Reply:
Correct - these are raw coordinates, and you’re right - they are not directly usable by LVGL.

A few other input drivers for: GT911, FT5336, CST816S also report raw coordinates.
To have a proper remapping, the user should perform a calibration, which would also account for small imperfections. Reducing the resolution beforehand can lead to worse results and is not necessary, so we think it should be done in the application layer and not by the driver itself.

We are open to suggestions if there is a better way to handle this, though.

Reply:
> Correct - these are raw coordinates, and you’re right - they are not directly usable by LVGL.
> 
> A few other input drivers for: GT911, FT5336, CST816S also report raw coordinates. To have a proper remapping, the user should perform a calibration, which would also account for small imperfections. Reducing the resolution beforehand can lead to worse results and is not necessary, so we think it should be done in the application layer and not by the driver itself.

The three you mentioned are all capacity touchscreen controller, they are factory calibrated and the output is a pixel coordinate, at most it has to be flipped or rotated, which the lvgl input driver supports.

How are you using this right now (if you don't mind sharing)? I imagine not with LVGL.

Reply:
We use our own application that handles remapping. You are correct that lvgl does not support it.
Therefore we have changed the driver. It now has several parameters for remapping that can be configured:
* Screen width
* Screen height
* X offset (in screen pixels)
* Y offset (in screen pixels)

All 4 are optional, the default values for offsets are 0, and width/height is 4096 (raw TS resolution).
Please take another look.